### PR TITLE
(fix): Refactor TI

### PR DESCRIPTION
**Pull request by rutvijmehta-harness**

### CODE REVIEW

File: ti/config/config.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+type Cfg struct {
+	client       *client.HTTPClient
+	sourceBranch string
+	targetBranch string
+	commitBranch string
+	dataDir      string
+}
+
+func NewCfg(endpoint, token, accountID, orgID, projectID, pipelineID, buildID, stageID, repo, sha, commitLink,
```
Comment (shubham149) [inline ti/config/config.go:13]: Rename it to New since it is present in config package

File: ti/instrumentation/utils.go [hunk 2]
```diff
@@ -189,37 +185,14 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger)
...
-	} else if isManual {
-		target = config.CommitBranch
-		if target == "" {
-			return ti.SelectTestsResp{}, fmt.Errorf("commit branch is not set")
-		}
-	}
-
-	ticonfig, err := getTiConfig(workspace, fs)
+	fs filesystem.FileSystem, cfg *tiCfg.Cfg) (ti.SelectTestsResp, error) {
+	tiConfig, err := getTiConfig(workspace, fs)
```
Comment (shubham149) [inline ti/instrumentation/utils.go:189]: There are 2 ti config present now.

File: ti/callgraph/upload.go [hunk 3]
```diff
@@ -26,34 +26,19 @@ const (
...
-	source := cfg.SourceBranch
-	if source == "" {
-		return fmt.Errorf("source branch is not set")
-	}
-	target := cfg.TargetBranch
-	if target == "" {
-		return fmt.Errorf("target branch is not set")
-	}
-
 	encCg, err := encodeCg(fmt.Sprintf(cgDir, pipeline.SharedVolPath), log)
```
Comment (shubham149) [inline ti/callgraph/upload.go:35]: Change this sharedVolPath to use ti config one

Comment (rutvijmehta-harness) [reply ti/callgraph/upload.go]: Good catch. Done

File: ti/client/http.go [hunk 4]
```diff
@@ -87,13 +88,19 @@ type HTTPClient struct {
 
 // Write writes test results to the TI server
 func (c *HTTPClient) Write(ctx context.Context, stepID, report string, tests []*ti.TestCase) error {
+	if c.Endpoint == "" {
+		return ErrConfigMissing
```
[the remark this answers is not in the document]
Comment (rutvijmehta-harness) [reply ti/client/http.go]: Done

File: ti/client/http.go [hunk 5]
```diff
@@ -102,24 +109,46 @@ func (c *HTTPClient) DownloadLink(ctx context.Context, language, os, arch, frame
 
 // SelectTests returns a list of tests which should be run intelligently
 func (c *HTTPClient) SelectTests(ctx context.Context, stepID, source, target string, in *ti.SelectTestsReq) (ti.SelectTestsResp, error) {
-	path := fmt.Sprintf(testEndpoint, c.AccountID, c.OrgID, c.ProjectID, c.PipelineID, c.BuildID, c.StageID, stepID, c.Repo, c.Sha, source, target)
 	var resp ti.SelectTestsResp
+	if c.Endpoint == "" {
```
[the remark this answers is not in the document]
Comment (rutvijmehta-harness) [reply ti/client/http.go]: Done

File: ti/config/config.go (hunk 1 above)
Comment (rutvijmehta-harness) [reply ti/config/config.go]: Done

File: ti/instrumentation/utils.go (hunk 2 above)
Comment (rutvijmehta-harness) [reply ti/instrumentation/utils.go]: Switched this to tiConfigYaml

**Review by shubham149 — APPROVED** [review]

**Review by shubham149 — APPROVED** [review]

**Pull request merged**